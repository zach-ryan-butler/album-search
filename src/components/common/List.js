import React from 'react';
import PropTypes from 'prop-types';

function List({ arr, Comp }) {
  const artistList = arr.map(obj => (
    <li key={obj.id}>
      <Comp props={obj} />
    </li>
  ));

  return (
    <ul>
      {artistList}
    </ul>
  );
}

List.propTypes = {
  arr: PropTypes.array.isRequired,
  Comp: PropTypes.func.isRequired
};

export default List;
