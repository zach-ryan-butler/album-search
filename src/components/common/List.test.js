import React from 'react';
import { shallow } from 'enzyme';
import List from './List';

describe('List component', () => {
  it('renders List', () => {
    const wrapper = shallow(<List
      arr={[]}
      Comp={() => {}}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
