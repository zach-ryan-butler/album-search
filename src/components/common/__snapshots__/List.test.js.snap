// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders List 1`] = `<ul />`;
